usl_pipeline/usl_lib/usl_lib/storage/metastore.py

```diff
@@ -155,6 +155,18 @@ def update_chunk_info(
             },
         )
 
+    @staticmethod
+    def delete_if_exists(db: firestore.Client, study_area_name: str) -> None:
+        """Deletes a given study area document if it exists.
+
+        Args:
+          db: The firestore database client to use.
+          study_area_name: The study area to delete.
+        """
+        ref = db.collection(STUDY_AREAS).document(study_area_name)
```

Comment:
deleting a document doesn't delete its subcollections
https://firebase.google.com/docs/firestore/manage-data/delete-data
so we'd need to delete the chunks subcollection as well.

Reply:
Oh, that's perfect catch! Thanks! Switched to list-all-and-delete-each strategy.